src/runtime.jl

```diff
@@ -216,15 +216,15 @@ function submit_task(f::Function, args::Tuple, kwargs::NamedTuple=NamedTuple();
         ""
     end
 
-    oid = GC.@preserve task_args begin
+    oids = GC.@preserve task_args begin
         ray_jll._submit_task(fd,
                              transform_task_args(task_args),
                              serialized_runtime_env_info,
                              resources)
     end
     # CoreWorker::SubmitTask calls TaskManager::AddPendingTask which initializes
     # the local ref count to 1, so we don't need to do that here.
-    return ObjectRef(oid; add_local_ref=false)
+    return ObjectRef(only(oids); add_local_ref=false)
```

Comment:
I'm unsure what we want to do here...

The changes so far implement "support for mulitple return variables" on the C++ side but the Julia-side remains as is and collects the `task_executor` result into a single `ObjectRef`.

i.e, in [`task_executor`](https://github.com/beacon-biosignals/Ray.jl/blob/05faae20661897adb79750400f06a6e892ec18e1/src/runtime.jl#L342C2-L343) we convert the function `result` to a `RayObject` and then `push!` this to the `returns` vector. This is completely agnostic to the type of the `result` and will always return a length-1 vector. 

The open question is what does "support multiple return variables" look like on the Julia-side?, i.e. do we want to retain current behaviour or do we want to loop over collections of `results` and `push!` them indvididually to the `returns` vector?  I'm not sure what advantages the latter gives us, and one potential downside is type instability...